src/components/Nav.js

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react';
+import ASGARDEO_LOGO from '../images/asgardeo-logo-transparent.png';
+import USER_LOGO from '../images/user.png';
+import routesConfig from '../configs/routes-config';
+import { useNavigate } from 'react-router-dom';
+import { useAuthContext } from '@asgardeo/auth-react';
+
+/**
+ * Nav bar component.
+ */
+const Nav = () => {
+    const { state, signOut, getDecodedIDToken } = useAuthContext();
+    const [ isResourcesAllowed, setIsResourcesAllowed ] = useState();
+
+    const navigate = useNavigate();
+
+    // Route to profile page.
+    const routeProfileChange = () =>{ 
+      let path = routesConfig.profile; 
+      navigate(path);
+      addActive();
+    }
+
+    // Route to API Call page.
+    const routeResourcesChange = () =>{ 
+      let path = routesConfig.resource; 
+      navigate(path);
+      addActive();
+    }
+
+    // Route to settings page.
+    const routeSettingsChange = () =>{ 
+      let path = routesConfig.settings; 
+      navigate(path);
+      addActive();
+    }
+
+    // Route to home page.
+    const routeHomeChange = () =>{ 
+      let path = routesConfig.home; 
+      navigate(path);
+      addActive();
+    }
+
+    // Filter the display of API Call section based on the application role.
+    useEffect(() => {
+      getDecodedIDToken().then((decodedIdToken) => {
+        if (decodedIdToken?.application_roles === "React-App-Manager") {
+          setIsResourcesAllowed(true);
+        }
+      }).catch((error) => {
+          
+      })
+    }, [getDecodedIDToken, state]);
+
+    // Add active class to the current button.
+    useEffect (() => {
+      addActive();
+    }, [ isResourcesAllowed ]);
+
+    // Add active class to the current button.
+    useEffect(() => {
+      const currentUrl = window.location.href;
+      const currentTabName = currentUrl.split("/")[3].replace("#", "");
+
+      var resourceTab = document.getElementById("resource");
```

Comment:
Why are we using `var` here?
IMO, we should be able to do this with the help of `react-router` utils.

Reply:
Fixed by 26f1298f704596c096081371805c1949b59a3277.